File: fortran_codes/bisection.f90
program bisection   
    
  real :: xl = 3., xu = 4., tol = 1e-4, eta = 100., xm(2)
  integer :: n = 0

  do while (eta > tol)

    xm(2) = (xu + xl)/2.

    if (func(xm(1))*func(xu) > 0) then 
      xu = xm(2)
    else 
      xl = xm(2)
    end if

    if (n > 0) then
      eta = abs ((xm(2) - xm(1))*100/xm(1))
    end if

      xm(1) = xm(2)
      n = n + 1

  end do


  print *, "The root of the given expression is :", xm(1)
  print *, "The tolerance is                    :", eta
    
end program bisection

real function func (x)
  real :: x 
  func =  4*exp(-x)*(3.2*sin(x) - 0.5*cos(x))
end function func
    

File: cpp_codes/bisection.cc
#include <iostream>
#include <cmath>

using namespace std;

double func (double x) {
  double func = 4*exp(-x)*(3.2*sin(x) - 0.5*cos(x));
  return func;
}

int main() {

  double xl = 3., xu = 4., tol = 1e-4, eta = 100., xm[2];
  int n = 0;

  while (eta > tol) {

    xm[1] = (xu + xl)/2.;

    if (func(xm[1])*func(xu) > 0) {
      xu = xm[1];
    } else {
      xl = xm[1];
    }

    if (n) {
      eta = abs ((xm[1] - xm[0])*100/xm[0]);
    }

    xm[0] = xm[1];
    n++;
  }

  cout << "The root of the given expression is : " << xm[1] << endl << "The tolerance is : " << eta << endl;
}